0.preprocess-sites/8.visualize-plate-layout.py

```diff
@@ -0,0 +1,392 @@
+    + gg.facet_wrap("~Well")
+    + gg.coord_fixed()
+    + gg.theme_bw()
+    + gg.theme(
+        axis_text=gg.element_blank(),
+        axis_title=gg.element_blank(),
+        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
+    )
+    + gg.scale_fill_cmap(name="magma")
+)
+output_file = pathlib.Path(output_figuresdir, "plate_layout_cells_count_per_well.png")
+by_well_gg.save(output_file, dpi=300, verbose=False)
+
+# Plot cell category ratios per well
+ratio_df = pd.pivot_table(
+    cell_count_df,
+    values="cell_count",
+    index=["site", "Plate", "Well", "Site", "x_loc", "y_loc"],
+    columns=["Cell_Quality"],
+)
+ratio_df["Sum"] = ratio_df.sum(axis=1)
+ratio_df["Pass_Filter"] = ratio_df[cell_filter].sum(axis=1)
+fail_filter = [cat for cat in cell_category_order if cat not in cell_filter]
+ratio_df["Fail_Filter"] = ratio_df[fail_filter].sum(axis=1)
+fail_filter_noempty = fail_filter.remove("Empty")
+ratio_df["Fail_Filter_noempty"] = ratio_df[fail_filter].sum(axis=1)
+ratio_df["Pass/Fail_withempty"] = ratio_df["Pass_Filter"] / ratio_df["Fail_Filter"]
+ratio_df["Pass/Fail_0empty"] = ratio_df["Pass_Filter"] / ratio_df["Fail_Filter_noempty"]
+ratio_df["Percent_Empty"] = ratio_df["Empty"] / ratio_df["Sum"]
+ratio_df = ratio_df.drop(
+    cell_category_order + ["Sum", "Pass_Filter", "Fail_Filter", "Fail_Filter_noempty"],
+    1,
+)
+ratio_df = ratio_df.stack().to_frame().reset_index().rename(columns={0: "value"})
+
+ratio_gg = (
+    gg.ggplot(ratio_df, gg.aes(x="x_loc", y="y_loc"))
+    + gg.geom_point(gg.aes(fill="value"), size=10)
+    + gg.geom_text(gg.aes(label="Site"), color="lightgrey")
+    + gg.facet_grid("Cell_Quality~Well", scales="free_y")
+    + gg.coord_fixed()
+    + gg.theme_bw()
+    + gg.theme(
+        axis_text=gg.element_blank(),
+        axis_title=gg.element_blank(),
+        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
+    )
+    + gg.scale_fill_cmap(name="magma")
+)
+
+output_file = pathlib.Path(output_figuresdir, "plate_layout_ratios_per_well.png")
+ratio_gg.save(output_file, dpi=300, verbose=False)
+
+# Create image dataframe
+sites = [x for x in os.listdir(batch_dir) if x not in ignore_files]
+image_list = []
+
+for site in sites:
+    try:
+        print(f"Now processing {site}...")
+        image_file = pathlib.Path(batch_dir, site, "Image.csv")
+        # Aggregates image information by site into single list
+        image_df = pd.read_csv(image_file)
+        image_df["site"] = site
+        image_df = image_df.assign(
+            Plate=[x[0] for x in image_df.site.str.split("-")],
+            Well=[x[1] for x in image_df.site.str.split("-")],
+            Site=[x[2] for x in image_df.site.str.split("-")],
+        )
+        image_list.append(image_df)
+    except FileNotFoundError:
+        print(f"{site} data not found")
+        continue
+
+image_df = pd.concat(image_list, axis="rows").reset_index(drop=True)
+print("Done concatenating image files")
```

Comment:
Another candidate to pull out into a function. Are we going to want to load image metadata in other places too?

Reply:
My gut says that the image metadata would only be used for QC so if we make this the QC recipe step (instead of a plate visualization step as I suggest in the bottommost comment), we wouldn't need it again. 
The one scenario I can think of is if we use outputs from QC to make a filter that we use to filter out sites for profiling. But that's not a thing as of now (and could perhaps even be written into this same recipe step).